buildbot.mariadb.org/master-libvirt/master.cfg

```diff
@@ -235,88 +187,101 @@ f_rpm_upgrade.addStep(getScript('rpm-upgrade.sh'))
-      properties={'systemdCapability': 'yes', 'needsGalera': 'yes', 'version_name': '35', 'arch': 'fedora35-amd64'},
-      factory=f_rpm_install))
+# Add the workers and builds based on the configured install builders (see constants.py)
+for builder_name in builders_install:
+    # Parse builder name
+    platform, os_name, os_version, builder_type = builder_name.split('-')[:4]
+
+    assert builder_type in ['rpm', 'deb']
+    assert platform in ['amd64', 'aarch64']
+
+    os_info_name = os_name + '-' + os_version
+    if platform == 'amd64':
+        libvirt_worker_name = 'bb-hz-bbw5-' + os_name + '-' + os_version
+        connection_url = 'qemu+ssh://buildbot@100.64.100.20:65001/system'
+    elif platform == 'aarch64':
+        libvirt_worker_name = 'bb-eq-arm1-' + os_name + '-' + os_version
+        connection_url = 'qemu+ssh://buildbot@100.64.100.17/system'
+
+    image_path = '/var/libvirt/images/' + libvirt_worker_name
+
+    c['workers'].append(worker.LibVirtWorker(libvirt_worker_name,
+        config["private"]["worker_pass"]["libvirt"],
+        util.Connection(connection_url),
+        image_path, build_wait_timeout=0, max_builds=1))
+
+    if builder_type == 'deb':
+        factory_install = f_deb_install
+        factory_upgrade = f_deb_upgrade
+        build_arch = platform
+    elif builder_type == 'rpm':
+        factory_install = f_rpm_install
+        factory_upgrade = f_rpm_upgrade
+        build_arch = os_name + os_version + '-' + platform
+
+    c['builders'].append(
+        util.BuilderConfig(name=builder_name,
+            workernames=libvirt_worker_name,
+            tags=[os_name, builder_type, 'install', 'kvm'],
+            collapseRequests=True,
+            nextBuild=nextBuild,
+            canStartBuild=canStartBuild,
+            properties={
+                'systemdCapability': 'yes',
+                'needsGalera': 'yes',
+                'dist_name': os_name,
+                'version_name': os_info[os_info_name]['version_name'],
+                'arch': platform},
+            factory=factory_install))
+
+    # Add major upgrade builder
+    major_upgrade_name = '-'.join(builder_name.split('-')[:5]) + '-major-upgrade'
+
+    c['builders'].append(
+        util.BuilderConfig(name=major_upgrade_name,
+          workernames=libvirt_worker_name,
+          tags=[os_name, builder_type, 'upgrade', 'kvm'],
+          collapseRequests=True,
+          nextBuild=nextBuild,
+          canStartBuild=canStartBuild,
+          properties={
+              'systemdCapability': 'yes', 
+              'needsGalera': 'yes', 
+              'dist_name': os_name, 
+              'version_name': os_info[os_info_name]['version_name'],
+              'arch': platform, 
+              'test_mode': 'server', 
+              "test_type": "major"
+             },
+          factory=factory_upgrade))
+
+    # Add minor upgrade builder
+    minor_upgrade_name = '-'.join(builder_name.split('-')[:5]) + '-minor-upgrade'
```

Comment:
`[:5]` looks a little fragile.

Reply:
I don't have any other ideas right now. The naming structure of the install builders should remain unchanged. If that changes, there are several places that will fail, but if you have any suggestions, let me know

Reply:
I don't have a good enough picture to offer suggestions.